### PR TITLE
feat: bump spring boot(3.2.3) and JDK17 in 2-advanced example module …

### PR DESCRIPTION
**Pull request by syshenyao**
…& tools module

### CODE REVIEW

File: tools/embedded-zookeeper/pom.xml [hunk 1]
```diff
@@ -31,8 +31,8 @@
     <version>1.0-SNAPSHOT</version>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
```
Comment (AlbumenJ) [inline tools/embedded-zookeeper/pom.xml:35]: Upgrade the version here is not a good idea now

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: tools/embedded-zookeeper/pom.xml (hunk 1 above)
Comment (AlbumenJ) [reply tools/embedded-zookeeper/pom.xml:35]: Do not change it in this PR

Comment (syshenyao) [reply tools/embedded-zookeeper/pom.xml:35]: Then I will roll back the modifications in this place, and what about the rest?

Comment (syshenyao) [reply tools/embedded-zookeeper/pom.xml]: done

**Pull request merged**